src/perf/best-practices.md

```diff
@@ -0,0 +1,411 @@
+---
+title: Performance best practices
+short-title: Best practices
+description: How to ensure that your Flutter app is performant.
+---
+
+{% include docs/performance.md %}
+
+Generally, Flutter applications are performant by default,
+so you only need to avoid common pitfalls to get excellent
+performance. These best practice recommendations will help you
+write the most performant Flutter app possible.
+
+Sometimes, apps are janky (not smooth) on some platforms.
+We have found two scenarios that can cause jankiness,
```

Comment:
I worry that saying "We have found two scenarios that can cause jankiness" makes it sound like there are only two reasons your app could be janky, which is not true. Maybe we should not call out saveLayer() and excessive layout passes here at all, but just allow users to read about them below? It seems odd that only these two scenarios get the special treatment of being called out here and in their own section below.

Reply:
Perhaps something like `Some rendering and layout operations are known to be slow, but cannot always be avoided. They should be used thoughtfully, following the guidance below` or something like that?

Reply:
Love it! Done